### ISSUE TITLE
Question regarding Voltron

### ISSUE THREAD

**Issue by goldbird5** [intuitive-robots/mdt_policy#13]
Hi, I have some questions regarding Voltron used for MDT-V.
Your Voltron code seems it's freezing Voltron's parameters during training.
(by using `self.vcond.eval()`, `params.requires_grad=False`, `with torch.no_grad()` in voltron_encoder.py)
However, why is the optimizer of 'mdtv_agent.py' containing Voltron's parameters, i.e. `self.img_encoder.parameters()`?

Besides, when I tried to train with `self.img_encoder.parameters()` commented out, it showed slightly decreased performance.
Do you have any idea regrading these?

Thank you.

**Comment by mbreuss** [comment]
Hi, in the training logs at the beginning it should say how many parameters are trained? Does it change when you modify the code snippet for that?

**Comment by goldbird5** [comment]
It seems yes to me. Both cases show same logs as follows.

<img width="456" alt="Image" src="https://github.com/user-attachments/assets/1891ad29-3092-4784-a525-c206fdc52117" />

Does this mean code snippet `self.img_encoder.parameters()` is just a simple mistake?
I'm curious about what was your intention, freeze or fintune the Voltron?


But, I don't have any idea where the performance difference comes from.

original MDT-V: until task5: 56.0% / avg_len: 3.774
commented MDT-V: until task5: 50.6% / avg_len: 3.572
(with seed 242 as default, I didn't test with another seed yet. I will let you know there are new results)

**Comment by mbreuss** [comment]
it could be some spread regarding seed performance but good catch, maybe train for 5 epochs longer and see how the performance changes.
I did not experiment a lot with various pretrained ViT models and I think that Voltron was a lucky pick for CALVIN and LIBERO. Other work uses MAE instead of  Voltron with good success, so if you want to improve the model I would give the MAE from Gr-1 and Seer a chance. 
Since you are working with MDT maybe this is also interesting for you:I will release my new codebase for my current project this week that surpasses MDT by a good margin on all CALVIN setups, in case you want to use the best policy available for your project.

**Comment by goldbird5** [comment]
Thanks for your advice regarding image encoders, and it's great to hear your new project.
I will check what I can do with those later.
Thank you:)
